### ISSUE TITLE
race on disconnect causes it to hang forever

### ISSUE THREAD

**Issue by yazug** [jasonrbriggs/stomp.py#160]
found a race

> conn = stomp.Connection([(args.host, args.port)])
> conn.start()                                                                                                                                                                                                   > conn.connect(login=args.user, passcode=args.password)
> conn.send(body=body, headers=headers, destination=args.destination)
> conn.disconnect()                                                 
> print 'Message sent.'

never reaches message sent 

but if you insert a time.sleep(2) right before conn.disconnect() it completes right away.  So it looks like there is some sort of race on disconnect.  

If you need more details to re-produce I can dig more on what is happening.

**Comment by yazug** [comment]
I saw this in a virtualenv

$ pip list
pip (1.4.1)
setuptools (0.9.8)
stomp.py (4.1.18)
wsgiref (0.1.2)

**Comment by yazug** [comment]
might be fixed by https://github.com/jasonrbriggs/stomp.py/pull/156

**Comment by ultimatecoder** [comment]
I am getting the exact behavior as @yazug is mentioning. I have tested with fetching latest head of master which is [this](https://github.com/jasonrbriggs/stomp.py/commit/25dd07c2b97a6fe93a3f09f9582698d59486a177) but still getting the same error. I think the #156 is fixing some other issue and not this.

**Comment by jasonrbriggs** [comment]
Try:

```
conn.connect(login=args.user, passcode=args.password, wait=True)
```

**Comment by jasonrbriggs** [comment]
Also added a fix for this to version 4.1.19 (so that it doesn't at least hang if you don't have `wait=True`)
